### DISCUSSION TITLE
Override app.html

### DISCUSSION THREAD

**Discussion by ghost** [sveltejs/kit#7906] [General]
How can you override app.html for a specific path? I assume hooks is the go to here, but havent figured it out. For instance if you want /routes/clean.txt/+page.svelte to not inherit any code from app.html but only contain whats actually generated by that +page.svelte file

**Comment by david-plugge** [comment]
You probably want to use `+server.js` files for files that are not html (files that are used inside your app)

**Reply by ghost** [reply]
Thank you. To try explain better, Im trying to produce a .xml file as then output (domain.com/file.xml) , and ideally would want to use +page.server.js or similar to generate it similar as it does for a +page.svelte, but override the app.html html elements. I realise there is a concept in this Ive not fully understood.

**Reply by david-plugge** [reply]
Something like this should do the trick

```ts
// src/routes/clean.xml/+server.js

/** @type {import('./$types').RequestHandler */
export function GET({ url }) {
	const content = `

<note>
	<to>Tove</to>
	<from>Jani</from>
	<heading>Reminder</heading>
	<body>Don't forget me this weekend!</body>
</note>
	
	`.trim();

	return new Response(content, {
		headers: { 'Content-Type': 'text/xml' }
	});
}
```

**Reply by ghost** [reply]
Thank you. That is brilliant.
